src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/BackedEnumValueResolver.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Attempt to resolve backed enum cases from request attributes, for a route path parameter,
+ * leading to a 404 Not Found if the attribute value isn't a valid backing value for the enum type.
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+class BackedEnumValueResolver implements ArgumentValueResolverInterface
+{
+    public function supports(Request $request, ArgumentMetadata $argument): bool
+    {
+        if (!is_subclass_of($argument->getType(), \BackedEnum::class)) {
+            return false;
+        }
+
+        if ($argument->isVariadic()) {
+            // only target route path parameters, which cannot be variadic.
+            return false;
+        }
+
+        // do not support if no value can be resolved at all
+        // letting the \Symfony\Component\HttpKernel\Controller\ArgumentResolver\DefaultValueResolver be used
+        // or \Symfony\Component\HttpKernel\Controller\ArgumentResolver fail with a meaningful error.
+        return $request->attributes->has($argument->getName());
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        $value = $request->attributes->get($argument->getName());
+
+        if (null === $value) {
+            yield null;
+
+            return;
+        }
+
+        if (!\is_int($value) && !\is_string($value)) {
+            throw new \LogicException(sprintf('Could not resolve the "%s $%s" controller argument: expecting an int or string, got %s.', $argument->getType(), $argument->getName(), get_debug_type($value)));
+        }
+
+        /** @var class-string<\BackedEnum> $enumType */
```

Comment:
What should we do about this Psalm error?

```php
UndefinedDocblockClass
src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/BackedEnumValueResolver.php:59:9: 
UndefinedDocblockClass: Docblock-defined class, interface or enum named BackedEnum does not exist (see https://psalm.dev/200)
```
I added an `UndefinedDocblockClass` [section in psalm.xml](https://github.com/symfony/symfony/pull/44831/commits/c5eeca7d4e17f68a926cdab9538cd3f53091c5ca) to fix it  for now (should be done in a dedicated PR targeting 4.4 if that's we want).

But would it be an issue to upgrade the Psalm workflow to use PHP 8.1?

Reply:
psalm is just a helper, it defines no goals at all. the only thing we need to do when seeing psalm errors is think about them and decide if it's applicable to the situation or not. So, what should we do? Analyze, and ignore here.

Reply:
Agreed, but at least adding adding this section to the 4.4 branch as already done for `UndefinedClass` would prevent further questions about this topic. Right? I'll open the PR.